### DISCUSSION TITLE
A way to access the Request object globally

### DISCUSSION THREAD

**Discussion by ccrvlh** [fastapi/fastapi#4799] [Questions] [answered]
We currently implement access control through decorators. 
We prefer this method since it keeps the route arguments clean, and makes it for a more explicit and easier way (in our opinion) to understand whats going on when comparing to `Depends`.

So normally we do something like this:

```python
@route.get('/health')
@access_control(superuser=True)
async def health_check():
    return {"health": True}
```

The challenge here is that `@access_control` needs something to work with, normally a header with the `Authorization` token.
But so far, from what we've seen, the only way to do that is to explicitly pass it to the route.

```python
@route.get('/health')
@access_control(superuser=True)
async def health_check(authorization: str = Header(None)):
    return {"health": True}
```

But `authorization` is never accessed in the route itself (we usually don't have any business logic on the routes, nor database access, it all goes through a `service`), it's only access in the `access_control` decorator with `kwargs`. This turns into a headache for linters, since they all complain the variable is not accessed. 

It would be great to have FastAPI to automatically pass the request as `kwargs` on the route. This would be implicit in the route itself, but would make for explicitly getting the information, somewhat similar to what Flask does. I'm normally not a big fan of implicit stuff, but in this case, I think it makes sense to have the request available inside a route, I mean, makes sense to have the context.

I guess it could be possible to do something like this with a Custom Middleware? But I didn't understand how would I inject the request without passing to the route.

Ideas?
Thanks!

**Comment by YuriiMotov** [comment, marked as answer]
In your `access_control` decorator you need to modify the signature of endpoint function by adding `request: Request` dependency.

You can take the inspiration from this comment: https://github.com/fastapi/fastapi/discussions/6285#discussioncomment-13459632
